server/server.go

```diff
@@ -531,6 +530,121 @@ func (j *jsonRPCHub) ApplyTxn(
 	return
 }
 
+func (j *jsonRPCHub) TraceMinedBlock(
+	block *types.Block,
+	tracer runtime.Tracer,
+) ([]interface{}, error) {
+	if block.Number() == 0 {
+		return nil, errors.New("genesis block can't have transaction")
+	}
+
+	parentHeader, ok := j.GetHeaderByHash(block.ParentHash())
+	if !ok {
+		return nil, errors.New("parent header not found")
+	}
+
+	blockCreator, err := j.GetConsensus().GetBlockCreator(block.Header)
+	if err != nil {
+		return nil, err
+	}
+
+	transition, err := j.BeginTxn(parentHeader.StateRoot, block.Header, blockCreator)
+	if err != nil {
+		return nil, err
+	}
+
+	transition.SetTracer(tracer)
+
+	results := make([]interface{}, len(block.Transactions))
+
+	for idx, tx := range block.Transactions {
+		tracer.Clear()
+
+		if _, err := transition.Apply(tx); err != nil {
+			return nil, err
+		}
+
+		results[idx] = tracer.GetResult()
+	}
+
+	return results, nil
+}
+
+func (j *jsonRPCHub) TraceMinedTxn(
+	block *types.Block,
+	targetTxHash types.Hash,
+	tracer runtime.Tracer,
+) (interface{}, error) {
+	if block.Number() == 0 {
+		return nil, errors.New("genesis block can't have transaction")
+	}
+
+	parentHeader, ok := j.GetHeaderByHash(block.ParentHash())
+	if !ok {
+		return nil, errors.New("parent header not found")
+	}
+
+	blockCreator, err := j.GetConsensus().GetBlockCreator(block.Header)
+	if err != nil {
+		return nil, err
+	}
+
+	transition, err := j.BeginTxn(parentHeader.StateRoot, block.Header, blockCreator)
+	if err != nil {
+		return nil, err
+	}
+
+	var targetTx *types.Transaction
+
+	for _, tx := range block.Transactions {
+		if tx.Hash == targetTxHash {
+			targetTx = tx
+
+			break
+		}
+
+		if _, err := transition.Apply(tx); err != nil {
+			return nil, err
+		}
+	}
+
+	if targetTx == nil {
+		return nil, errors.New("target tx not found")
+	}
+
+	transition.SetTracer(tracer)
+
+	if _, err := transition.Apply(targetTx); err != nil {
+		return nil, err
+	}
+
+	return tracer.GetResult(), nil
+}
+
+func (j *jsonRPCHub) TraceCall(
+	tx *types.Transaction,
```

Comment:
This is not correct. You cannot use core `Transaction` and `Header` types for encoding/decoding the jsonrpc call itself. The same applies for the other functions.

Reply:
My mistake here, I thought this was the endpoint specification.